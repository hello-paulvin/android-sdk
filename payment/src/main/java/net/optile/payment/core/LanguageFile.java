/*
 * Copyright(c) 2012-2018 optile GmbH. All Rights Reserved.
 * https://www.optile.net
 *
 * This software is the property of optile GmbH. Distribution  of  this
 * software without agreement in writing is strictly prohibited.
 *
 * This software may not be copied, used or distributed unless agreement
 * has been received in full.
 */

package net.optile.payment.core;

import net.optile.payment.model.Interaction;

import java.util.Properties;

/**
 * Class holding the language entries for the payment page, ApplicableNetwork or AccountRegistration
 */
public final class LanguageFile {

    public final static String KEY_BUTTON_DATE = "button.update.label";
    public final static String KEY_AUTO_REGISTRATION = "autoRegistrationLabel";
    public final static String KEY_ALLOW_RECURRENCE = "allowRecurrenceLabel";

    private final Properties lang;

    /**
     * Construct an empty LanguageFile
     */
    public LanguageFile() {
        this.lang = new Properties();
    }

    public String translate(String key) {
        return translate(key, null);
    }

    public String translate(String key, String defValue) {
        return key != null ? lang.getProperty(key, defValue) : defValue;
    }

    public String translateError(String error) {
        return translate("error.".concat(error));
    }

    public String translateAccount(String account) {
        return translate("account.".concat(account).concat(".label"));
    }

    public String translateInteraction(Interaction interaction) {
        String key = "interaction.".concat(interaction.getCode()).concat(".").concat(interaction.getReason());
        return translate(key);
    }

    public Properties getProperties() {
        return lang;
    }
}
