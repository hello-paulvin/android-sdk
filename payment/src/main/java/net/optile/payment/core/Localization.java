/*
 * Copyright (c) 2019 optile GmbH
 * https://www.optile.net
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package net.optile.payment.core;

import java.util.HashMap;
import java.util.Map;
import java.util.Properties;

import android.content.Context;
import android.text.TextUtils;
import net.optile.payment.R;
import net.optile.payment.model.Interaction;

/**
 * Class holding the localization files and local translations.
 * If the translation is not available in a localization file with the given name, it will try to obtain the translation from the shared file.
 * If the shared file is not set or does not contain the translation, then it will try to obtain the translation from the local translations map.
 */
public final class Localization {

    // all entries starting with pmlocal originate from the local strings.xml file
    public final static String AUTO_REGISTRATION = "autoRegistrationLabel";
    public final static String ALLOW_RECURRENCE = "allowRecurrenceLabel";
    public final static String NETWORK_LABEL = "network.label";

    public final static String BUTTON_BACK = "button.back.label";
    public final static String BUTTON_CANCEL = "pmlocal.button.cancel";
    public final static String BUTTON_UPDATE = "pmlocal.button.update";
    public final static String BUTTON_RETRY = "pmlocal.button.retry";

    public final static String LIST_TITLE = "pmlocal.list.title";
    public final static String LIST_PRESET_TEXT = "networks.preset.text";
    public final static String LIST_HEADER_PRESET = "networks.preset.title";
    public final static String LIST_HEADER_SAVEDACCOUNTS = "savedAccountsLabel";
    public final static String LIST_HEADER_OTHERACCOUNTS = "addNewAccountLabel";
    public final static String LIST_HEADER_NETWORKS = "pmlocal.list.header.networks";

    public final static String CHARGE_TITLE = "pmlocal.charge.title";
    public final static String CHARGE_TEXT = "pmlocal.charge.text";
    public final static String CHARGE_INTERRUPTED = "pmlocal.charge.interupted";

    public final static String ERROR_CONNECTION = "pmlocal.error.connection";
    public final static String ERROR_DEFAULT = "pmlocal.error.default";

    public final static String ACCOUNTHINT_TITLE = "title";
    public final static String ACCOUNTHINT_TEXT = "text";

    private final Map<String, Properties> files;
    private final Map<String, String> localTranslations;
    private Properties sharedFile;

    private Localization() {
        this.files = new HashMap<>();
        this.localTranslations = new HashMap<>();
    }

    /**
     * Get the instance of this Localization
     *
     * @return the instance of this localization
     */
    public static Localization getInstance() {
        return InstanceHolder.INSTANCE;
    }

    /**
     * Helper method to check if a translation exist for the account hint.
     *
     * @param fileName the name of the Properties file
     * @param account name of the account
     * @return true if it has a translation for the account hint, false otherwise
     */
    public static boolean hasAccountHint(String fileName, String account) {
        String val = translateAccountHint(fileName, account, ACCOUNTHINT_TITLE);
        return !TextUtils.isEmpty(val);
    }

    /**
     * Helper method to obtain the translation for the error from the specified file.
     *
     * @param fileName name of the localization file
     * @param error the error which should be translated
     * @return the translation or null if not found
     */
    public static String translateError(String fileName, String error) {
        return translate(fileName, "error." + error, null);
    }

    /**
     * Helper method to obtain the translation for the interaction.
     *
     * @param interaction to be translated
     * @return translation or null if not found
     */
    public static String translateInteraction(Interaction interaction) {
        String key = "interaction." + interaction.getCode() + "." + interaction.getReason();
        return getInstance().getTranslation(key, null);
    }

    /**
     * Helper method to obtain the translation for the account label.
     *
     * @param fileName name of the localization file
     * @param account name of the account
     * @return translation or null if not found
     */
    public static String translateAccountLabel(String fileName, String account) {
        return translate(fileName, "account." + account + ".label", null);
    }

    /**
     * Helper method to obtain the translation of the account hint with the given type.
     *
     * @param fileName name of the localization file
     * @param account name of the account
     * @param type of the account hint, either ACCOUNTHINT_TEXT or ACCOUNTHINT_TITLE
     * @return the translation or null if not found
     */
    public static String translateAccountHint(String fileName, String account, String type) {
        String key = "account." + account + "." + "hint." + "where." + type;
        return translate(fileName, key, null);
    }

    /**
     * Helper method to obtain the translation for the given key from the shared file or local translations map.
     *
     * @param key of the translation
     * @return the translation or null if not found
     */
    public static String translate(String key) {
        return getInstance().getTranslation(key, null);
    }

    /**
     * Helper method to obtain the translation for the given key from the localization file.
     *
     * @param fileName name of the localization file
     * @param key of the translation
     * @return the translation or null if not found
     */
    public static String translate(String fileName, String key) {
        return translate(fileName, key, null);
    }

    /**
     * Helper method to obtain the translation for the given key from the localization file.
     * If the translation could not be found then return the defaultValue.
     *
     * @param fileName name of the localization file
     * @param key of the translation
     * @param defaultValue returned when the translation could not be found
     * @return the translation or the defaultValue if the translation was not found
     */
    public static String translate(String fileName, String key, String defaultValue) {
        return getInstance().getTranslation(fileName, key, defaultValue);
    }

    /**
     * Get the translation from the localization file with the given name.
     * If the file does not contain the translation then try to obtain it from the shared file.
     * If the shared file does not exist or does not contain the translation then try to obtain the translation from the local translations map.
     *
     * @param fileName name of the localization file
     * @param key of the translation
     * @param defaultValue returned when the translation could not be found
     * @return the translation or defaultValue if the translation was not found
     */
    public String getTranslation(String fileName, String key, String defaultValue) {
        if (!TextUtils.isEmpty(fileName)) {
            String value = getValue(files.get(fileName), key);
            if (!TextUtils.isEmpty(value)) {
                return value;
            }
        }
        return getTranslation(key, defaultValue);
    }

    /**
     * Get the translation from the shared localization file.
     * If the shared file does not exist or does not contain the translation then try to obtain the translation from the local translations map.
     *
     * @param key of the translation
     * @param defaultValue returned when the translation could not be found
     * @return the translation or defaultValue if the translation was not found
     */
    public String getTranslation(String key, String defaultValue) {
        String value = getValue(sharedFile, key);
        if (TextUtils.isEmpty(value)) {
            value = localTranslations.get(key);
        }
        return TextUtils.isEmpty(value) ? defaultValue : value;
    }

    /**
     * Load the local translations into this Localization Object.
     * The local translations serve as a fallback when the translation could not be found in any of the files.
     *
     * @param context used to load the local translations
     */
    public void loadLocalTranslations(Context context) {
        if (localTranslations.size() != 0) {
            return;
        }
        localTranslations.put(BUTTON_CANCEL, context.getString(R.string.pmlocal_button_cancel));
        localTranslations.put(BUTTON_RETRY, context.getString(R.string.pmlocal_button_retry));
        localTranslations.put(BUTTON_UPDATE, context.getString(R.string.pmlocal_button_update));
        localTranslations.put(LIST_TITLE, context.getString(R.string.pmlocal_list_title));
        localTranslations.put(LIST_HEADER_NETWORKS, context.getString(R.string.pmlocal_list_header_networks));
        localTranslations.put(CHARGE_TITLE, context.getString(R.string.pmlocal_charge_title));
        localTranslations.put(CHARGE_TEXT, context.getString(R.string.pmlocal_charge_text));
        localTranslations.put(CHARGE_INTERRUPTED, context.getString(R.string.pmlocal_charge_interrupted));
        localTranslations.put(ERROR_CONNECTION, context.getString(R.string.pmlocal_error_connection));
        localTranslations.put(ERROR_DEFAULT, context.getString(R.string.pmlocal_error_default));
    }

    public void clear() {
        files.clear();
        localTranslations.clear();
        sharedFile = null;
    }

    public void setSharedFile(Properties sharedFile) {
        this.sharedFile = sharedFile;
    }

    public boolean hasSharedFile() {
        return sharedFile != null;
    }

    public void putFile(String fileName, Properties properties) {
        files.put(fileName, properties);
    }

    public boolean hasFile(String fileName) {
        return files.containsKey(fileName);
    }

    private String getValue(Properties prop, String key) {
        return prop != null ? prop.getProperty(key) : null;
    }

    private static class InstanceHolder {
        static final Localization INSTANCE = new Localization();
    }
}
