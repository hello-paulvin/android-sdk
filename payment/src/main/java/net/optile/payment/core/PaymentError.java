/*
 * Copyright (c) 2019 optile GmbH
 * https://www.optile.net
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package net.optile.payment.core;

import com.google.gson.JsonSyntaxException;

import android.os.Parcel;
import android.os.Parcelable;
import android.text.TextUtils;
import android.util.Log;
import net.optile.payment.model.ErrorInfo;
import net.optile.payment.util.GsonHelper;

/**
 * A class representing the details about the error
 */
public final class PaymentError implements Parcelable {

    public final static Parcelable.Creator<PaymentError> CREATOR = new Parcelable.Creator<PaymentError>() {

        public PaymentError createFromParcel(Parcel in) {
            return new PaymentError(in);
        }

        public PaymentError[] newArray(int size) {
            return new PaymentError[size];
        }
    };

    /**
     * The optional network status code like 404 or 500
     */
    private int statusCode;
    /**
     * The optional message containing details about the error
     */
    private String message;
    /**
     * The optional ErrorInfo received from the payment Api
     */
    private ErrorInfo errorInfo;
    /**
     * The optional cause of the internal error
     */
    private Throwable cause;
    /**
     * This optional flag to indicate that this error was caused by a network failure
     */
    private boolean networkFailure;

    /**
     * Construct a new PaymentError containing the message what went wrong
     *
     * @param message the error message
     */
    public PaymentError(final String message) {
        this.message = message;
    }

    /**
     * Construct a new PaymentError containing the network failure information
     *
     * @param statusCode the status code
     * @param message the error message
     */
    public PaymentError(final int statusCode, final String message) {
        this.statusCode = statusCode;
        this.message = message;
    }

    /**
     * Construct a new PaymentError containing the network failure information
     *
     * @param statusCode the status code
     * @param errorInfo the error info
     */
    public PaymentError(final int statusCode, final ErrorInfo errorInfo) {
        this.statusCode = statusCode;
        this.message = errorInfo.getResultInfo();
        this.errorInfo = errorInfo;
    }

    /**
     * Construct a new PaymentError containing the cause of the internal error
     *
     * @param cause the cause of the internal error
     */
    public PaymentError(final Throwable cause) {
        this.message = cause.getMessage();
        this.cause = cause;
    }

    /**
     * Construct a new PaymentError containing information what caused the error
     *
     * @param message the error message
     * @param cause the cause of the internal error
     */
    public PaymentError(final String message, final Throwable cause) {
        this.message = message;
        this.cause = cause;
    }

    /**
     * Construct a new PaymentError containing information what caused the error
     * A network failure may be caused by an IOException.
     *
     * @param cause the cause of the internal error
     * @param networkFailure indicating if this error was caused by a network failure
     */
    public PaymentError(final Throwable cause, final boolean networkFailure) {
        this.message = cause.getMessage();
        this.cause = cause;
        this.networkFailure = networkFailure;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public String getMessage() {
        return message;
    }

    public ErrorInfo getErrorInfo() {
        return errorInfo;
    }

    public Throwable getCause() {
        return cause;
    }

    public boolean getNetworkFailure() {
        return networkFailure;
    }

    private PaymentError() {
    }

    private PaymentError(Parcel in) {
        this.statusCode = in.readInt();
        this.message = in.readString();

        GsonHelper gson = GsonHelper.getInstance();
        try {
            String json = in.readString();
            if (!TextUtils.isEmpty(json)) {
                this.errorInfo = gson.fromJson(json, ErrorInfo.class);
            }
        } catch (JsonSyntaxException e) {
            // this should never happen since we use the same GsonHelper
            // to produce these Json strings
            Log.w("pay", e);
            throw new RuntimeException(e);
        }
        this.cause = (Throwable) in.readSerializable();
        this.networkFailure = in.readInt() == 1;
    }


    /**
     * Helper class to construct a new InternalError from the given Throwable
     *
     * @param cause of the error
     * @return the InternalError containing detailed information about the error
     */
    public static PaymentError fromThrowable(Throwable cause) {
        if (cause instanceof PaymentException) {
            return ((PaymentException) cause).error;
        }
        return new PaymentError(cause);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public int describeContents() {
        return 0;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void writeToParcel(Parcel out, int flags) {
        out.writeInt(statusCode);
        out.writeString(message);

        GsonHelper gson = GsonHelper.getInstance();
        String errorInfoJson = null;
        if (errorInfo != null) {
            errorInfoJson = gson.toJson(errorInfo);
        }
        out.writeString(errorInfoJson);
        out.writeSerializable(cause);
        out.writeInt(networkFailure ? 1 : 0);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String toString() {

        final StringBuilder sb = new StringBuilder();
        sb.append("PaymentError[");

        if (statusCode != 0) {
            sb.append(" statusCode: ");
            sb.append(statusCode);
        }
        if (message != null) {
            sb.append(" message: ");
            sb.append(message);
        }
        if (cause != null) {
            sb.append(" cause: ");
            sb.append(cause.toString());
        }
        sb.append(" networkFailure: ");
        sb.append(networkFailure);
        sb.append("]");
        return sb.toString();
    }
}
