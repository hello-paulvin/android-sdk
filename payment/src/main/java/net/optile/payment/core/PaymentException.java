/*
 * Copyright (c) 2019 optile GmbH
 * https://www.optile.net
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package net.optile.payment.core;

/**
 * The PaymentException containing the payment error details
 */
public class PaymentException extends Exception {

    /**
     * The details of the error
     */
    public final PaymentError error;

    /**
     * {@inheritDoc}
     *
     * @param message the extra error info
     */
    public PaymentException(final String message) {
        super(message);
        this.error = new PaymentError(message);
    }

    /**
     * {@inheritDoc}
     *
     * @param cause the cause
     */
    public PaymentException(final Throwable cause) {
        super(cause.getMessage(), cause);
        this.error = new PaymentError(cause);
    }
    
    /**
     * {@inheritDoc}
     *
     * @param message the error message
     * @param cause the cause
     */
    public PaymentException(final String message, final Throwable cause) {
        super(message, cause);
        this.error = new PaymentError(message, cause);
    }

    /**
     * {@inheritDoc}
     *
     * @param error the internal error
     */
    public PaymentException(final PaymentError error) {
        super(error.getMessage(), error.getCause());
        this.error = error;
    }
}
