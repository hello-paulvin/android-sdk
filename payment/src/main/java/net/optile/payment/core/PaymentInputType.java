/*
 * Copyright(c) 2012-2018 optile GmbH. All Rights Reserved.
 * https://www.optile.net
 *
 * This software is the property of optile GmbH. Distribution  of  this
 * software without agreement in writing is strictly prohibited.
 *
 * This software may not be copied, used or distributed unless agreement
 * has been received in full.
 */

package net.optile.payment.core;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import android.support.annotation.StringDef;

/**
 * Class containing the different payment input types i.e. number, iban, bic etc.
 */
public class PaymentInputType {

    public final static String ACCOUNT_NUMBER = "number";
    public final static String HOLDER_NAME = "holderName";
    public final static String EXPIRY_DATE = "expiryDate";
    public final static String EXPIRY_MONTH = "expiryMonth";
    public final static String EXPIRY_YEAR = "expiryYear";
    public final static String VERIFICATION_CODE = "verificationCode";
    public final static String BANK_CODE = "bankCode";
    public final static String IBAN = "iban";
    public final static String BIC = "bic";

    /**
     * Check if the given type is a valid payment input type
     *
     * @param type the payment input type to validate
     * @return true when valid, false otherwise
     */
    public static boolean isValid(final String type) {

        if (type != null) {
            switch (type) {
            case ACCOUNT_NUMBER:
            case HOLDER_NAME:
            case EXPIRY_DATE:
            case EXPIRY_MONTH:
            case EXPIRY_YEAR:
            case VERIFICATION_CODE:
            case BANK_CODE:
            case IBAN:
            case BIC:                
                return true;
            }
        }
        return false;
    }

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
            ACCOUNT_NUMBER,
                HOLDER_NAME,
                EXPIRY_DATE,
                EXPIRY_MONTH,
                EXPIRY_YEAR,
                VERIFICATION_CODE,
                BANK_CODE,
                IBAN,
                BIC
        })
    public @interface Definition { }
}

