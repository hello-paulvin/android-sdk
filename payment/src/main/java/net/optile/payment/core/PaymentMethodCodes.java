/*
 * Copyright (c) 2019 optile GmbH
 * https://www.optile.net
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package net.optile.payment.core;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import android.support.annotation.StringDef;

/**
 * Class containing PaymentMethodCodes for which the accountNumber and verificationCodes are custom validated.
 * Custom validation settings are defined in the R.raw.validations file.
 */
public class PaymentMethodCodes {

    public final static String AMEX = "AMEX";
    public final static String CASTORAMA = "CASTORAMA";
    public final static String DINERS = "DINERS";
    public final static String DISCOVER = "DISCOVER";
    public final static String MASTERCARD = "MASTERCARD";
    public final static String UNIONPAY = "UNIONPAY";
    public final static String VISA = "VISA";
    public final static String VISA_DANKORT = "VISA_DANKORT";
    public final static String VISAELECTRON = "VISAELECTRON";
    public final static String CARTEBANCAIRE = "CARTEBANCAIRE";
    public final static String MAESTRO = "MAESTRO";
    public final static String MAESTROUK = "MAESTROUK";
    public final static String POSTEPAY = "POSTEPAY";
    public final static String SEPADD = "SEPADD";
    public final static String JCB = "JCB";    

    /**
     * Check if the given type is a valid payment input type
     *
     * @param type the payment input type to validate
     * @return true when valid, false otherwise
     */
    public static boolean isValid(final String type) {

        if (type != null) {
            switch (type) {
                case AMEX:
                case CASTORAMA:
                case DINERS:
                case DISCOVER:
                case MASTERCARD:
                case UNIONPAY:
                case VISA:
                case VISA_DANKORT:
                case VISAELECTRON:
                case CARTEBANCAIRE:
                case MAESTRO:
                case MAESTROUK:
                case POSTEPAY:
                case SEPADD:
                case JCB:
                    return true;
            }
        }
        return false;
    }

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
        AMEX,
        CASTORAMA,
        DINERS,
        DISCOVER,
        MASTERCARD,
        UNIONPAY,
        VISA,
        VISA_DANKORT,
        VISAELECTRON,
        CARTEBANCAIRE,
        MAESTRO,
        MAESTROUK,
        POSTEPAY,
        SEPADD,
        JCB
    })
    public @interface Definition { }
}
