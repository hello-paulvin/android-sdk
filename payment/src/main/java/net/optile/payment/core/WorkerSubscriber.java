/*
 * Copyright(c) 2012-2018 optile GmbH. All Rights Reserved.
 * https://www.optile.net
 *
 * This software is the property of optile GmbH. Distribution  of  this
 * software without agreement in writing is strictly prohibited.
 *
 * This software may not be copied, used or distributed unless agreement
 * has been received in full.
 */

package net.optile.payment.core;

/**
 * WorkerSubscriber listening for a WorkerTask to be completed
 */
public interface WorkerSubscriber<V> {

    /**
     * The WorkerTask has successfully completed
     *
     * @param param the result parameter
     */
    void onSuccess(V param);

    /**
     * An error occurred while executing the WorkerTask
     *
     * @param error the cause of the error
     */
    void onError(Throwable error);
}
