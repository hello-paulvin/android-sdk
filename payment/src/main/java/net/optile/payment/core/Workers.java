/*
 * Copyright(c) 2012-2018 optile GmbH. All Rights Reserved.
 * https://www.optile.net
 *
 * This software is the property of optile GmbH. Distribution  of  this
 * software without agreement in writing is strictly prohibited.
 *
 * This software may not be copied, used or distributed unless agreement
 * has been received in full.
 */

package net.optile.payment.core;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * The Singleton class maintaining a list of Workers processing background tasks
 */
public final class Workers {

    private ExecutorService networkService;

    private Workers() {
        this.networkService = Executors.newCachedThreadPool();
    }

    /**
     * Get the instance of this Workers
     *
     * @return the instance of this Workers
     */
    public static Workers getInstance() {
        return InstanceHolder.INSTANCE;
    }

    /**
     * Get the WorkerExecutor for network tasks
     *
     * @return the network executor for network tasks
     */
    public ExecutorService forNetworkTasks() {
        return networkService;
    }

    private static class InstanceHolder {
        static final Workers INSTANCE = new Workers();
    }
}
