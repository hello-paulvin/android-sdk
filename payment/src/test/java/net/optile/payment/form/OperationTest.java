/*
 * Copyright (c) 2019 optile GmbH
 * https://www.optile.net
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package net.optile.payment.form;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;
import static org.junit.Assert.assertNotNull;

import java.net.MalformedURLException;
import java.net.URL;

import org.json.JSONException;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import net.optile.payment.core.PaymentException;
import net.optile.payment.core.PaymentInputType;

@RunWith(RobolectricTestRunner.class)
public class OperationTest {

    @BeforeClass
    public static void beforeAll() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }

    @Test(expected = IllegalArgumentException.class)
    public void putValue_invalidName_exception() throws PaymentException {
        Operation operation = new Operation("VISA", "CREDIT_CARD", "CHARGE", createTestURL());
        operation.putValue(null, "Foo");
    }

    @Test
    public void putValue_success() throws PaymentException, JSONException {
        Operation operation = new Operation("VISA", "CREDIT_CARD", "CHARGE", createTestURL());
        operation.putValue(PaymentInputType.ACCOUNT_NUMBER, "accountnumber123");
        operation.putValue(PaymentInputType.HOLDER_NAME, "John Doe");
        operation.putValue(PaymentInputType.EXPIRY_MONTH, 12);
        operation.putValue(PaymentInputType.EXPIRY_YEAR, 2019);
        operation.putValue(PaymentInputType.VERIFICATION_CODE, "123");
        operation.putValue(PaymentInputType.BANK_CODE, "bankcode123");
        operation.putValue(PaymentInputType.IBAN, "iban123");
        operation.putValue(PaymentInputType.BIC, "bic123");
        operation.putValue(PaymentInputType.ALLOW_RECURRENCE, "true");
        operation.putValue(PaymentInputType.AUTO_REGISTRATION, "true");
        expect(operation.toJson()).toMatchSnapshot();
    }

    private URL createTestURL() {
        URL url = null;

        try {
            url = new URL("http://localhost/charge");
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        assertNotNull(url);
        return url;
    }
}

