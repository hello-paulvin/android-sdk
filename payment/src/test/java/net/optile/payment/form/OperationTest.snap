net.optile.payment.form.OperationTest.putValue_success=[
  "{\"allowRecurrence\":\"true\",\"autoRegistration\":\"true\",\"account\":{\"number\":\"accountnumber123\",\"holderName\":\"John Doe\",\"expiryMonth\":12,\"expiryYear\":2019,\"verificationCode\":\"123\",\"bankCode\":\"bankcode123\",\"iban\":\"iban123\",\"bic\":\"bic123\"}}"
]